cloud/azure/main.tf

```diff
@@ -75,127 +55,60 @@ locals {
 
-  container_network_interface {
-    name = "civiform-server-network-interface"
+  # Define Linux as Host OS
+  kind     = "Linux"
+  reserved = true # Mandatory for Linux plans
 
-    ip_configuration {
-      name      = "civiform-network-interface-ip-configuration"
-      subnet_id = azurerm_subnet.server_subnet.id
-    }
+  # Choose size
+  sku {
+    tier     = "Standard"
+    size     = "S1"
+    capacity = "2"
   }
 }
 
-resource "azurerm_container_group" "cg" {
-  name                = "civiform-container-group"
+resource "azurerm_app_service" "civiform-app" {
+  name                = var.application_name
   location            = azurerm_resource_group.rg.location
   resource_group_name = azurerm_resource_group.rg.name
-  network_profile_id  = azurerm_network_profile.civiform_network_profile.id
-  ip_address_type     = "Private"
-  os_type             = "Linux"
-
-  container {
-    name   = "dh-civiform"
-    image  = "civiform/civiform:latest"
-    cpu    = "0.5"
-    memory = "1.5"
-
-    environment_variables = {
-      # Azure specifically requires the username to be in password@hostname format. Note
-      # the hostname in this case is not the fully qualified hostname.
-      DB_USERNAME    = "${azurerm_postgresql_server.civiform.administrator_login}@${azurerm_postgresql_server.civiform.name}"
-      DB_PASSWORD    = azurerm_postgresql_server.civiform.administrator_login_password
-      DB_JDBC_STRING = "jdbc:postgresql://${azurerm_postgresql_server.civiform.fqdn}:5432/postgres?ssl=true&sslmode=require"
-
-      SECRET_KEY = "insecure-secret-key"
-    }
+  app_service_plan_id = azurerm_app_service_plan.plan.id
+  app_settings = {
+    WEBSITES_ENABLE_APP_SERVICE_STORAGE = false
 
-    # ports exposed on the individual containers
-    ports {
-      port     = 80
-      protocol = "TCP"
-    }
-    ports {
-      port     = 443
-      protocol = "TCP"
-    }
-  }
 
-  # ports exposed on the container group
-  exposed_port {
-    port     = 80
-    protocol = "TCP"
+    # Below only necessary if using private Docker repository
+    DOCKER_REGISTRY_SERVER_USERNAME = var.docker_username
+    DOCKER_REGISTRY_SERVER_PASSWORD = var.docker_password
+
+    DB_USERNAME    = "${azurerm_postgresql_server.civiform.administrator_login}@${azurerm_postgresql_server.civiform.name}"
+    DB_PASSWORD    = azurerm_postgresql_server.civiform.administrator_login_password
+    DB_JDBC_STRING = "jdbc:postgresql://${azurerm_postgresql_server.civiform.fqdn}:5432/postgres?ssl=true&sslmode=require"
+
+    SECRET_KEY = "insecure-secret-key"
   }
-  exposed_port {
-    port     = 443
-    protocol = "TCP"
+  # Configure Docker Image to load on start
+  site_config {
+    linux_fx_version                     = "DOCKER|${var.docker_username}/${var.docker_repository_name}:${var.tag_name}"
+    always_on                            = "true"
+    acr_use_managed_identity_credentials = "true"
+    vnet_route_all_enabled               = "true"
```

Comment:
question (non-blocking): do the values of these arguments need to be strings or can they be booleans?

Reply:
changed them to booleans and it seems to still work